python/ql/lib/semmle/python/frameworks/ServerLess.qll

```diff
@@ -0,0 +1,59 @@
+import python
+import codeql.serverless.ServerLess
+import semmle.python.dataflow.new.DataFlow
+import semmle.python.dataflow.new.RemoteFlowSources
+
+private module YamlImpl implements Input {
+  import semmle.python.Files
+  import semmle.python.Yaml
+}
+
+module SL = ServerLess<YamlImpl>;
+
+/**
+ * Gets a function that is a serverless request handler.
+ *
+ * For example: if an AWS serverless resource contains the following properties (in the "template.yml" file):
+ * ```yaml
+ * Handler: mylibrary.handler
+ * Runtime: pythonXXX
+ * CodeUri: backend/src/
+ * ```
+ *
+ * And a file "mylibrary.py" exists in the folder "backend/src" (relative to the "template.yml" file).
+ * Then the result of this predicate is a function exported as "handler" from "mylibrary.py".
+ * The "mylibrary.py" file could for example look like:
+ *
+ * ```python
+ * def handler(event):
+ *  ...
+ * ```
+ */
+private Function getAServerlessHandler() {
+  exists(File file, string stem, string handler, string runtime, Module mod |
+    SL::hasServerlessHandler(stem, handler, _, runtime) and
+    file.getAbsolutePath() = stem + ".py" and
+    // if runtime is specified, it should be python
+    (runtime = "" or runtime.matches("python%"))
+  |
+    mod.getFile() = file and
+    mod.getAnExport() = handler and
+    result.getEnclosingModule() = mod and
+    result.getName() = handler
```

Comment:
`mod.getAnExport()` uses points-to under the hook, so let's not use that.

How about just the following?

```suggestion
    mod.getFile() = file and
    result.getScope() = mod and
    result.getName() = handler
```

then we don't wrongly mark both `handler` functions below as serverless handlers

```py
def handler(event, context): pass

class Foo:
    def handler(self): pass
```

(we're restricted to top-level function definitions though, which on top of my head seems ok)